crates/cairo-profiler/src/profile_builder.rs

```diff
@@ -267,3 +276,27 @@ pub fn build_profile(samples: &[Sample]) -> pprof::Profile {
         default_sample_type: 0,
     }
 }
+
+pub fn save_profile(target_path: &Utf8PathBuf, profile: &pprof::Profile) -> Result<()> {
+    if let Some(parent) = target_path.parent() {
+        fs::create_dir_all(parent)
+            .context("Failed to create parent directories for the output file")?;
+    }
+    let mut file = fs::File::create(target_path).context("Failed to create the output file")?;
+
+    let mut buffer = BytesMut::new();
+    profile
+        .encode(&mut buffer)
+        .expect("Failed to encode the profile to the buffer");
```

Comment:
Is there any guide lines when do we use here `expect` and when `result`? I some times find this inconsistent

```suggestion
        .context("Failed to encode the profile to the buffer")?;
```

Reply:
My rule of thumb is:
- is something unexpected? did you expect it should work? `.expect()`, cuz this is potentially the bug in our code, so we panic
- is something somewhat expected? possibly user induced in some way, or something the user has a control of? `.context()`, as an error (with additional context) suffices

But I know that this is pretty mixed in the profiler codebase. Maybe one day I take it upon myself and put it all in order :D

Reply:
Yep, I have a similar approach. I just wanted to ask so we are on the same page. I guess this is a valid case for using `expect`.

Maybe this should also be an `expect`, as we expect the reading buffer to pass:

```rust
encoder
.read_to_end(&mut encoded_buffer)
.context("Failed to read bytes from the encoder")?;
```

Nevertheless, it's such a minor thing that I'll probably stop making these kinds of comments, as they don't bring too much value to the PRs.

Reply:
thanks for pointing this out btw! ❤️